### PR TITLE
User Registration Captcha

### PR DESCRIPTION
**Pull request by brianteeman**
Pull Request for Issue #21921 .

### Summary of Changes
Moves the captcha to the end of the form. This had already been done for contacts and content

### Testing Instructions
1. Enable recaptcha
2. set the captcha in global config
3. enable user registration
4. create custom user fields for registration
5. check the registration form on the front end - captcha displayed
6. disable captcha in global config
7. check the registration form on the front end - no captcha displayed
8. enable captcha in the users component
9.  check the registration form on the front end - captcha displayed

### Actual result BEFORE applying this Pull Request
the captcha is rendered after the default fields but before any custom fields

### Expected result AFTER applying this Pull Request
the captcha is rendered at the end of the form

### CODE REVIEW

File: components/com_users/forms/registration.xml [hunk 1]
```diff
@@ -73,12 +73,15 @@
...
 		/>
+	</fieldset>
 
+	<fieldset name="captcha">
 		<field
 			name="captcha"
 			type="captcha"
 			label="COM_USERS_CAPTCHA_LABEL"
 			validate="captcha"
+			namespace="contact"
```
Comment (alikon) [inline components/com_users/forms/registration.xml]: shouldn't be something else than contact  ?

Comment (brianteeman) [reply components/com_users/forms/registration.xml]: corry - copy paste error

**Comment by crystalenka** [comment]
Is this reading for testing?

**Comment by brianteeman** [comment]
> Is this ready for testing?

of course. its not a draft - just no one seems interested in testing it and would prefer to complain about the bug

**Comment by crystalenka** [comment]
I have tested this item :white_check_mark: successfully on 75e0fbe902a1d3adace2c75d89710d0e88b698bf<br /><br />Works as expected - CAPTCHA shown before custom fields before the PR, and shown after custom fields after the PR was applied. Thanks!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38604">issues.joomla.org/tracker/joomla-cms/38604</a>.</sub>

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 75e0fbe902a1d3adace2c75d89710d0e88b698bf<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38604">issues.joomla.org/tracker/joomla-cms/38604</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38604">issues.joomla.org/tracker/joomla-cms/38604</a>.</sub>

**Comment by brianteeman** [comment]
Anything preventing this bug being merged?

**Comment by HLeithner** [comment]
thanks

**Comment by HLeithner** [comment]
Thanks, I rebased this on 4.3 because it's not a bugfix.

**Comment by HLeithner** [comment]
I added the migration documentation at https://github.com/joomla/Manual/pull/52

**Comment by brianteeman** [comment]
of course its a bug fix

**Comment by crystalenka** [comment]
Would agree this is a bug fix rather than a feature. Captcha should always be rendered right before submit. Even if it wasn't technically "broken" it was a usability bug.

**Comment by brianteeman** [comment]
@crystalenka seems that @hleithner thinks anything that doesn't spit out an error message is not a bug even if it doesnt do what it says it will do. another example https://github.com/joomla/joomla-cms/pull/38953#issuecomment-1288060275

**Comment by HLeithner** [comment]
it's not me alone that thinks like this but if you need to blame a person then pick me happy with it.

**Comment by brianteeman** [comment]
Well you are the only one who commented so if anyone else does they are anonymous and it would be impossible for me to know.

The fact still remains that "you" are not being consistent in deciding what is a bug and what is a feature. It just depends on who created the pr.

Looks like we are returning to the bad old days where functionality that doesn't work is never fixed because to fix it is a new feature. And only code from "real" developers is acceptable in this elitist world.

Even when code is contributed specifically at the request of one of these "real" developers it is ignored because the wrong type of person contributed it.

**Comment by HLeithner** [comment]
Doesn't matter who provided the PR, maybe you noticed that this PR has been merged already and about 40 other PRs this weekend from various people so please stop attacking me and tell me that I prefer some persons more then other.

thanks

**Comment by bembelimen** [comment]
> Looks like we are returning to the bad old days where functionality that doesn't work is never fixed because to fix it is a new feature. And only code from "real" developers is acceptable in this elitist world.

I just checked the PR and for me it looks like the code is fixed...?

**Comment by laoneo** [comment]
@brianteeman you can also add me to the list of persons who are cautious about merging pr's into a patch branch. Patch releases are really there to stabilize a minor version and should only receive bug fixes (mainly regressions) and not cosmetic stuff. This doesn't mean that they are not relevant, but they should be added to a .0 version where people expect a different behavior, especially in the UI. This is common software development practice and not something Harald and I have invented.

**Comment by brianteeman** [comment]
This was not a cosmetic fix. This was a bug fix. This should have been fixed in 4.2

**Pull request merged**